### PR TITLE
UIOR-5 Assign tags to Order Line Records

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
To display, assign, unassign, and output tags associated with individual ORDER LINE records within the FOLIO Orders app.
https://issues.folio.org/browse/UIOR-5
## Approach
Use `Tags` from `stripes-acq-components`
## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/OrderLinesList/Details/OrderLineDetails.js [hunk 1]
```diff
@@ -1,32 +1,35 @@
...
-      type: 'okapi',
-      path: `${ORDERS_API}/%{orderId}`,
-      throwErrors: false,
-      fetch: false,
-      accumulate: true,
+      ...baseManifest,
+      path: (queryParams, pathComponents, resourceData, logger, props) => {
+        const orderId = get(props, ['resources', 'orderLine', 'records', 0, 'purchaseOrderId']);
+
+        return `${ORDERS_API}/${orderId}`;
```
Comment (aliaksei-chumakou) [inline src/OrderLinesList/Details/OrderLineDetails.js]: `orderId` might be null or undefined, right?

Comment (AndreiShumski2) [reply src/OrderLinesList/Details/OrderLineDetails.js]: Yep. Fixed

**Comment by aliaksei-chumakou** [comment]
@AndreiShumski2 I actually wonder what is the reason of the whole `order` manifest change?

**Comment by AndreiShumski2** [comment]
> @AndreiShumski2 I actually wonder what is the reason of the whole `order` manifest change?
The main reason is to update `OrderLineDetails` component every time when the tags list is changed. Previously it depended on `this.props.match.params.id` only.

**Comment by aliaksei-chumakou** [comment]
@AndreiShumski2  but `order` doesn't change

**Comment by AndreiShumski2** [comment]
> @AndreiShumski2 but `order` doesn't change

Yes. but previously order line and order changed only if `this.props.match.params.id` is changed. But if you add or remove tags, `OrderLineDetails` component doesn't rerender because id wasn't changed. So, in this case, I changed manifest for order as we did in organizations and invoices modules to avoid using additional manifest for `orderId`

File: src/OrderLinesList/Details/OrderLineDetails.js [hunk 2]
```diff
@@ -1,118 +1,163 @@
...
+  const lineId = match.params.id;
+  const line = get(resources, ['orderLine', 'records', 0], {});
+  const order = get(resources, ['order', 'records', 0], {});
+
+  const fetchLineDetails = useCallback(
+    () => {
+      mutator.orderLine.reset();
+      mutator.orderLine.GET();
+      mutator.order.reset();
+      mutator.orderLine.GET()
```
Comment (aliaksei-chumakou) [inline src/OrderLinesList/Details/OrderLineDetails.js]: the same as above (might be wrong mutator)

File: src/OrderLinesList/Details/OrderLineDetails.js [hunk 3]
```diff
@@ -1,118 +1,163 @@
...
+        });
     },
-    orderLine: {
-      ...baseManifest,
-      path: `${LINES_API}/:{id}`,
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [lineId],
+  );
+
+  useEffect(() => {
```
Comment (aliaksei-chumakou) [inline src/OrderLinesList/Details/OrderLineDetails.js]: I think `useEffect` should be dependent on `match.params.id`

Comment (aliaksei-chumakou) [inline src/OrderLinesList/Details/OrderLineDetails.js]: `useEffect(fetchLineDetails, [lineId]);` ?

File: src/OrderLinesList/Details/OrderLineDetails.js (hunk 2 above)
Comment (AndreiShumski2) [reply src/OrderLinesList/Details/OrderLineDetails.js]: fixed

**Pull request merged**